crates/biome_service/src/workspace/server.rs

```diff
@@ -628,14 +694,13 @@ impl Workspace for WorkspaceServer {
         let project_key = params.project_key;
         let path = params.path.as_path();
         let capabilities = self.get_file_capabilities(&params.path);
-
         let handle = WorkspaceSettingsHandle::from(
             self.projects
                 .get_settings(project_key)
                 .ok_or_else(WorkspaceError::no_project)?,
         );
         let mut file_features = FileFeaturesResult::new();
-        let language = DocumentFileSource::from_path(path);
+        let language = self.get_file_source(path);
```

Comment:
`get_file_source()` is also called inside `get_file_capabilities()`.

Given that it's quite a heavy method, I think it makes sense to drop the call to `get_file_capabilities()` above, and just add this:

```suggestion
        let language = self.get_file_source(path);
        let capabilities = self.features.get_capabilities(path, language);
```